packages/atlas/src/components/_video/VideoPlayer/VideoPlayer.tsx

```diff
@@ -160,6 +160,8 @@ const VideoPlayerComponent: ForwardRefRenderFunction<HTMLVideoElement, VideoPlay
   const [isFullScreen, setIsFullScreen] = useState(false)
   const [isPiPEnabled, setIsPiPEnabled] = useState(false)
   const [isSettingsPopoverOpened, setIsSettingsPopoverOpened] = useState(false)
+  const [waitingStateCount, setWaitingStateCount] = useState(0)
+  const [slowNetworkEventSent, setSlowNetworkEventSent] = useState(false)
```

Comment:
Maybe we can go with `useRef` here? I or is there a reason for the player to re-render when each of these changes?